superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/CccsGrid.tsx

```diff
@@ -303,8 +338,8 @@ export default function CccsGrid({
         onRangeSelectionChanged={onRangeSelectionChanged}
         onSelectionChanged={onSelectionChanged}
         rowData={rowData}
-        paginationPageSize={pageSize.current}
-        pagination={pageSize.current !== 0}
```

Comment:
That makes sense. However, negative values (yes, it's allowed!) should also have no pagination. Actually, the built-in table does something funky. If the row size is set to 100, the page length is set to -99 and it will show 1 row and no pagination! I think pagination should determine if pagination is <= 0.